codice/ddf-jaxb
# Addressing concerns in #6

### Requirements
* Filling out the template is required. Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

@stustison @coyotesqrl 
@bakejeyner @blen-desta 

### Description of the Change

Cleaning up all the pom.xml files.

### Alternate Designs

Two main decisions were made while refactoring. First was to get all the generic configurations used by every child pom into the root pom. The second was to get as much use out of default configurations as possible provided by the `jaxb2-maven-plugin` as well as removing unnecessary tags along the way. An example of this is `<forceRegenerate>`. It has been deprecated as of version `0.12.1` of the plugin and claimed to be unneeded. However, there's a known issue where if multiple `<execution>` blocks are run with the same target directory then it will fail to generate any code past the first `<execution>` block if `<forceRegeneration>` is not present. The work around for the future to avoid using a deprecated tag is to use separate target directories which has been added in the latest commit. 

### Benefits

Maintainability and consistency. 

### Possible Drawbacks

None-

### Verification Process

Shamelessly copied from [here](https://github.com/codice/ddf/pull/3898)
1. Full [DDF](https://github.com/codice/ddf) CI build must pass
2. Install DDF and test affected functionality
   * Create CSW sources and ensure they work correctly
   * Create WFS 1.0, 1.1, and 2.0 sources and ensure they work correctly
   * Ensure security - CAS and XACML in particular - still work
   * Ensure that metacard transformations (ingests, etc.) still work correctly

### Applicable Issues

Fixes: #6
